src/QueryResultCsvWriter.php

```diff
@@ -15,60 +17,92 @@
  */
 class QueryResultCsvWriter
 {
+    public const MAX_VALUE_STATE_KEY = 'lastFetchedRow';
+
     protected array $state;
 
+    protected int $rowsCount;
+
+    protected ?array $lastRow;
+
     public function __construct(array $state)
     {
         $this->state = $state;
     }
 
     public function writeToCsv(QueryResult $result, ExportConfig $exportConfig, string $csvFilePath): ExportResult
```

Comment:
Tu sme mali jednu dlhu `writeToCsv` metodu, ktoru som tiez rozdelil na viacero mensich, kedze vacsinou chce programator upravit iba cast chovania, nie cele to prepisat.

Reply:
BTW Je to dobre pokryte unit testami: https://github.com/keboola/db-extractor-adapter/blob/master/tests/phpunit/AbstractExportAdapterTest.php